### PR TITLE
Tests written and method updated to fix the bug.

### PR DESCRIPTION
**Pull request by Tanvish**
Closes issue: #312 

I updated the show method as was specified on the issue page. I also wrote a test to check for the issue and it passed.

### CODE REVIEW

File: spec/controllers/application_submissions_spec.rb [hunk 1]
```diff
@@ -395,6 +395,16 @@
         expect(response).not_to render_template 'show'
       end
     end
+    context 'record does not belong in the directory' do
```
Comment (werebus) [inline spec/controllers/application_submissions_spec.rb]: This `context` description is a little confusing. The context here is that there is no current user at all. It doesn't really matter what the state of the record is. Maybe `'no current user'` or something to that effect?

**Comment by Tanvish** [comment]
Following up on my previous commit to seeing if there are any more changes required?

**Comment by Tanvish** [comment]
Following up on this pull request.

**Comment by werebus** [comment]
:question: 

Both @Anbranin and I have approved it. Do you have any outstanding questions?

**Comment by Tanvish** [comment]
Oh, no just that I am not authorized to merge this pull request for some reason.

**Comment by werebus** [comment]
Ah yes. Sorry.

Our repository permissions aren't entirely uniform. I think we've gone back and forth on whether trainees should be able to click the merge button.

**Comment by Tanvish** [comment]
Oh, alright.

**Pull request merged**